### ISSUE TITLE
Support generic methods

### ISSUE THREAD

**Issue by VladRassokhin** [sdaschner/jaxrs-analyzer#30]
For now (version 0.8) analyzer fails on methods like
```java
  public <T> T getSingletonService(@NotNull Class<T> clazz) {
    // Some code here
  }
```

With stacktrace like:
```stacktrace
Could not analyze class for type analysis: java.lang.RuntimeException: javassist.NotFoundException: T
java.lang.RuntimeException: java.lang.RuntimeException: javassist.NotFoundException: T
	at com.sebastian_daschner.jaxrs_analyzer.model.types.Type.<init>(Type.java:66)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.utils.JavaUtils.getReturnType(JavaUtils.java:217)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.mapGetter(TypeAnalyzer.java:219)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.lambda$analyzeClass$275(TypeAnalyzer.java:150)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeClass(TypeAnalyzer.java:151)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:121)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToArray(JsonMapper.java:168)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:116)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToObject(JsonMapper.java:144)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.lambda$analyzeClass$276(TypeAnalyzer.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeClass(TypeAnalyzer.java:151)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:121)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToArray(JsonMapper.java:168)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:116)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToObject(JsonMapper.java:144)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.lambda$analyzeClass$276(TypeAnalyzer.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeClass(TypeAnalyzer.java:151)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:121)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToObject(JsonMapper.java:144)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.lambda$analyzeClass$276(TypeAnalyzer.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeClass(TypeAnalyzer.java:151)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:121)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToArray(JsonMapper.java:168)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:116)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToObject(JsonMapper.java:144)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.lambda$analyzeClass$274(TypeAnalyzer.java:148)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeClass(TypeAnalyzer.java:148)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:121)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.JsonMapper.addToObject(JsonMapper.java:144)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.lambda$analyzeClass$274(TypeAnalyzer.java:148)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeClass(TypeAnalyzer.java:148)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyzeInternal(TypeAnalyzer.java:121)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.TypeAnalyzer.analyze(TypeAnalyzer.java:95)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.lambda$interpretResponse$270(ResultInterpreter.java:132)
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1540)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.interpretResponse(ResultInterpreter.java:130)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.lambda$interpretResourceMethod$265(ResultInterpreter.java:109)
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1540)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.interpretResourceMethod(ResultInterpreter.java:109)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.interpretMethodResult(ResultInterpreter.java:87)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.lambda$interpretClassResult$264(ResultInterpreter.java:69)
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1540)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.interpretClassResult(ResultInterpreter.java:69)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1540)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.results.ResultInterpreter.interpret(ResultInterpreter.java:55)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.ProjectAnalyzer.analyze(ProjectAnalyzer.java:86)
	at com.sebastian_daschner.jaxrs_analyzer.JAXRSAnalyzer.analyze(JAXRSAnalyzer.java:69)
	at com.sebastian_daschner.jaxrs_analyzer.Main.main(Main.java:131)
Caused by: java.lang.RuntimeException: javassist.NotFoundException: T
	at com.sebastian_daschner.jaxrs_analyzer.model.types.Type.<init>(Type.java:35)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.LinkedList$LLSpliterator.forEachRemaining(LinkedList.java:1235)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.sebastian_daschner.jaxrs_analyzer.model.types.TypeExtractor.toTypeParameters(TypeExtractor.java:52)
	at com.sebastian_daschner.jaxrs_analyzer.model.types.Type.<init>(Type.java:64)
	... 116 more
Caused by: javassist.NotFoundException: T
	at javassist.ClassPool.get(ClassPool.java:450)
	at com.sebastian_daschner.jaxrs_analyzer.model.types.TypeExtractor.toErasuredClass(TypeExtractor.java:39)
	at com.sebastian_daschner.jaxrs_analyzer.model.types.Type.<init>(Type.java:32)
	... 125 more
```

**Comment by VladRassokhin** [comment]
Related to #12

**Comment by jeff9finger** [comment]
Closed #31 as a duplicate of this one.

**Comment by sdaschner** [comment]
Resolved in c617231. Feel free to test the `0.9-SNAPSHOT` version in the Maven Central snapshot repo.

**Comment by jeff9finger** [comment]
Thanks for the fix.

The NotFoundException is resolved. But I am still getting errors (not exceptions). Will open a new issue for that.

**Comment by sdaschner** [comment]
FYI: just released version `0.9`.
You may want to update your dependencies ;-)
